task-sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -995,7 +999,7 @@ def _on_term(signum, frame):
         )
         state = TaskInstanceState.FAILED
         error = e
-    except (AirflowTaskTimeout, AirflowException, AirflowRuntimeError) as e:
+    except (AirflowTaskTimeout, AirflowException, AirflowRuntimeError, RuntimeError) as e:
```

Comment:
If we re-order this to after the `AirflowTaskTerminated` case, couldn't this be simplified to `except Exception`.

And now I look at this, I don't think this case is needed at all, and the `except BaseException as e:` we already have is enough?